pandas/core/indexes/datetimes.py

```diff
@@ -2128,19 +2128,34 @@ def month_name(self, locale=None):
 
     def day_name(self, locale=None):
         """
-        Return the day names of the DateTimeIndex with specified locale.
+        Return the day names.
+        
+        Provides day names of the DateTimeIndex for the dataframe which consists of the time and date data with specified locale.
 
         Parameters
         ----------
         locale : string, default None (English locale)
-            locale determining the language in which to return the day name
-
+            The locale determines the language in which the day name should be returned.
+         
         Returns
         -------
-        month_names : Index
-            Index of day names
-
-        .. versionadded:: 0.23.0
+        day_names : Index
+            Index (instance) of day names
+            
+        See Also
+        --------
+        month_name : Return the month names
+        
+        Examples
+        --------
+        #From the time stamps data based on it the day_name is returned. 
```

Comment:
Do you need help making examples? This should be like

```python
>>> idx = pd.date_range('2017', periods=2)
>>> idx
DatetimeIndex(['2017-01-01', '2017-01-02'],
              dtype='datetime64[ns]', freq='D')
>>> idx.day_name()
>>> idx.day_name()
Index(['Sunday', 'Monday'], dtype='object')
```

Then show an example with a different locale like

```python
In [37]: idx.day_name('zh_hk')
Out[37]: Index(['週日', '週一'], dtype='object')
```